packages/host/tests/helpers/adapter.ts

```diff
@@ -0,0 +1,302 @@
+import {
+  Loader,
+  LocalPath,
+  RealmAdapter,
+  RealmPaths,
+  baseRealm,
+  createResponse,
+} from '@cardstack/runtime-common';
+
+import {
+  FileRef,
+  Kind,
+  Realm,
+  TokenClaims,
+  UpdateEventData,
+} from '@cardstack/runtime-common/realm';
+
+import { WebMessageStream, messageCloseHandler } from './stream';
+
+import { Dir, createJWT, testRealmURL } from '.';
+
+type CardAPI = typeof import('https://cardstack.com/base/card-api');
+
+class TokenExpiredError extends Error {}
+class JsonWebTokenError extends Error {}
+
+interface TestAdapterContents {
+  [path: string]: string | object;
+}
+
+let shimmedModuleIndicator = '// this file is shimmed';
+
+export class TestRealmAdapter implements RealmAdapter {
+  #files: Dir = {};
+  #lastModified: Map<string, number> = new Map();
+  #paths: RealmPaths;
+  #subscriber: ((message: UpdateEventData) => void) | undefined;
+  #contents: TestAdapterContents = {};
+
+  constructor(realmURL = new URL(testRealmURL)) {
+    this.#paths = new RealmPaths(realmURL);
+  }
+
+  async setContents(contents: TestAdapterContents, loader: Loader) {
+    let now = Date.now();
+    let realmURL = this.#paths.url;
+    this.#contents = contents;
+
+    let cardApi = await loader.import<CardAPI>(`${baseRealm.url}card-api`);
+
+    for (const [path, value] of Object.entries(contents)) {
+      if (path.endsWith('.json') && cardApi.isCard(value)) {
+        value.id = `${realmURL}${path.replace(/\.json$/, '')}`;
+        cardApi.setCardAsSavedForTest(value);
+      }
+    }
+
+    for (const [path, mod] of Object.entries(contents)) {
+      if (path.endsWith('.gts') && typeof mod !== 'string') {
+        let moduleURLString = `${realmURL}${path.replace(/\.gts$/, '')}`;
+        loader.shimModule(moduleURLString, mod as object);
```

Comment:
Without this, serializing the card in the next step will fail because it will not be able to identify the card, which happens because there is no module present in the loader (specifically, `Loader.identify(card)` will return `undefined` in `identifyCard`

Reply:
I think having `loader.shimModule` sounds undesirable since we were trying to avoid this in the first place, but it looks like it's still needed for the purpose of serialization of the card for the flat files in test realm adapter.

However the change in `openFile` (passing the module value as a symbol and propagating to the loader) still fixes many failing scenarios when loaders are isolated (check the PR's description)